common/searchattribute/name_type_map.go

```diff
@@ -117,3 +118,13 @@ func (m NameTypeMap) IsDefined(name string) bool {
 	}
 	return false
 }
+
+func (m NameTypeMap) Copy() NameTypeMap {
```

Comment:
nit: s/Copy/Clone ?
Where are this and the `AddCustomSearchAttribute` below used?

Reply:
In `query_interceptors_test.go`.  I surmise there is a more elegant Go solution. I'm all ears to learn :)